### DISCUSSION TITLE
Is there a way to stop cloning or downloading from my public repositories?

### DISCUSSION THREAD

**Discussion by PN1019** [community/community#23248] [New to GitHub]
<p>Well, I don’t want my project public repo to be used without my permission.So is there any way I can restrict people to use my repo without my permission.I want people to take my permission before using my work.</p>

**Comment by PN1019** [comment]
<p>Hope this clears your doubt</p>
<blockquote>
<p><a class="mention" href="/stevencopeland">@stevencopeland</a> wrote:<br>
Could you please rephrase your question? I am not sure what you mean.</p>
</blockquote>
<p>:smiley:</p>

**Comment by winstliu** [comment, marked as answer]
<p>No, there is no way to prevent people from cloning or downloading your repository if they have access to it.  If you are worried about others using your code without your express permission, I would suggest adding a license.  I’ve personally found <a href="https://choosealicense.com/" rel="nofollow noopener">https://choosealicense.com/</a> to be helpful when deciding which license to use.</p>

**Reply by Divuzki** [reply]
It will be a very good features if it was added back to github

**Reply by ajneel** [reply]
This feature needs to be enabled as it brings high risk to the software development company or even teams in terms of employee turnover.

**Reply by SlothTheCoder** [reply]
Tell me, which license?

**Reply by BPowell76** [reply]
@ajneel Honestly, companies should be using an internal git server if they're worried about code being leaked to the public or forked or whatever. I did this at my job once our IT people expressed concern over storing code in a private github repo. Granted, I was storing the code on a personal account and not a corporate account.

**Comment by nadiajoyce** [comment]
<p><a class="mention" href="/pn1019">@pn1019</a> It looks like you forgot to include some content on this reply. Did you edit your original post? Or did you mean to include more in this comment?</p>

**Comment by ghost** [comment]
<p><a class="mention" href="/50wliu">@50wliu</a> suggested a great idea - <a href="https://github.community/t/is-there-a-way-to-stop-cloning-or-downloading-from-my-public-repositories/293/4">Use a license</a>. It protects you legally, and gives you control over your codebase…</p>
<p>If you’re determined to make it non-downloadable, you can always make it a private repository, but then you lose the factor of open-source-iness.</p>

**Reply by SlothTheCoder** [reply]
Which license?

**Comment by PN1019** [comment]
<p><a class="mention" href="/nadiajoyce">@nadiajoyce</a> I edited my original post. BDW, I got the query answered.</p>

**Comment by Darksidezoo** [comment]
<ol>
<li>
<p>ZIP or RAR your repository</p>
</li>
<li>
<p>Put it on google drive</p>
</li>
<li>
<p>Click get shareable link</p>
</li>
<li>
<p>Click advanced - Check prevent people from downloading or sharing the file</p>
</li>
</ol>
<p>Anyone with the link can see what’s inside the directories and the structure, but can’t open or view the content of the files. </p>

**Reply by SlothTheCoder** [reply]
About the alternative of google drive?

**Comment by smartarix** [comment]
<p>Is there any way for rejecting large clone requests?</p>
<p>I want to reject clone request if repo size is more than 2GB</p>

**Comment by MattG-DPDesign** [comment]
<p>Could you demonstrate how this would look like in Github? I’m new to web hosting and I don’t want teachers taking all my own work for their classes.</p>

**Reply by SlothTheCoder** [reply]
Oh my god.

**Comment by onexdata** [comment]
Is this a feature request somewhere? Now that we have codespaces, it seems logical to be able to disable cloning, and only allow users to clone to a codespace. This would be absolutely ideal and perfect for contractors, and even employees at companies who want high levels of security.

**Reply by ferdnyc** [reply]
Public repos are just that: Public.

GitHub offers private repos (for a fee), for users who want to limit access to authorized users. If you don't want other people to access to your code, why would you keep it in a public repo???

(Codespaces would hardly provide "high levels of security" for hosted code. When a repo is cloned into a codespace, liberating it from that codespace would be child's play for anyone with access and the ability to string together a half-dozen UNIX shell commands.)

This request is a bit like saying, "How do I prevent people from viewing the photos I published to my public web gallery?" You don't, because the whole **point** of publishing a public gallery is so that other people can view the photos!

**Reply by Kanishk1420** [reply]
When I share my project with prospective internship hosts, I need a view-only link that prevents cloning or downloading without explicit permission. If anyone does manage to download the code, I’d like to receive an immediate notification.

**Reply by ferdnyc** [reply]
OK, but you "needing" that doesn't change the fact that it's never going to happen.
